### PR TITLE
fix: buggy chapter segments rendering

### PR DESCRIPTION
**Pull request by luwes**
don't love this but I couldn't get it to work otherwise.

### CODE REVIEW

File: src/js/controller.js [hunk 1]
```diff
@@ -564,8 +564,12 @@ export const MediaUIStates = {
       return Array.from(chaptersTrack?.cues  ?? [])
         .map((/** @type VTTCue */{ text, startTime, endTime }) => ({ text, startTime, endTime }));
     },
-    mediaEvents: ['loadstart', 'loadedmetadata'],
-    textTracksEvents: ['addtrack', 'removetrack'],
+    mediaEvents: ['loadstart'],
+    textTracksEvents: ['addtrack', 'removetrack', 'change'],
```
Comment (luwes) [inline src/js/controller.js:568]: I added `change` as well because it makes sense to re-render when the active chapters track changes

Comment (cjpillsbury) [reply src/js/controller.js:568]: cool def makes sense to me.

File: src/js/controller.js [hunk 2]
```diff
@@ -565,7 +565,15 @@ export const MediaUIStates = {
         .map((/** @type VTTCue */{ text, startTime, endTime }) => ({ text, startTime, endTime }));
     },
     mediaEvents: ['loadstart', 'loadedmetadata'],
-    textTracksEvents: ['addtrack', 'removetrack'],
+    textTracksEvents: ['addtrack', 'removetrack', 'change'],
+    mediaSetCallback(media, callback) {
+      const chaptersTrack = media.querySelector('track[kind="chapters"][default][src]');
+      chaptersTrack?.addEventListener('load', callback);
+    },
+    mediaUnsetCallback(media, callback) {
```
Comment (cjpillsbury) [inline src/js/controller.js:573]: lol started peeping this before I got pulled away and looks like you anticipated the cleanup concern already. 🔥

**Review by cjpillsbury — APPROVED** [review]
Some concerns/tradeoffs with this approach, but we'll address those in a followup.

**Pull request merged**